pkg/ddc/vineyard/transform.go

```diff
@@ -109,10 +109,13 @@ func (e *VineyardEngine) transformMasters(runtime *datav1alpha1.VineyardRuntime,
 
 func (e *VineyardEngine) transformWorkers(runtime *datav1alpha1.VineyardRuntime, value *Vineyard) (err error) {
 	value.Worker = Worker{}
-	if runtime.Spec.Worker.Replicas == 0 {
-		value.Worker.Replicas = 1
-	} else {
+	// respect the replicas of runtime if the replicas of worker and runtime are both specified
+	if runtime.Spec.Replicas != 0 {
```

Comment:
I think if the user doesn't specify `runtime.Spec.Replicas`, it should be 0. I don't suggest using the default value as 1.

Reply:
Ok, thanks.